Patch by MSP-Greg
lengtens timeouts for Appveyor parallel testing
diff --git a/test/ruby/test_file_exhaustive.rb b/test/ruby/test_file_exhaustive.rb
index 4c5540eb4c..e3b59bc34b 100644
--- a/test/ruby/test_file_exhaustive.rb
+++ b/test/ruby/test_file_exhaustive.rb
@@ -1316,7 +1316,8 @@ def test_flock_exclusive
       assert_separately(["-rtimeout", "-", regular_file], "#{<<~begin}#{<<~"end;"}")
       begin
         open(ARGV[0], "r") do |f|
-          Timeout.timeout(0.1) do
+          time_out = !(/mingw|mswin|bccwin/ !~ RUBY_PLATFORM) ? 0.5 : 0.1
+          Timeout.timeout(time_out) do
             assert(!f.flock(File::LOCK_SH|File::LOCK_NB))
           end
         end
@@ -1324,8 +1325,9 @@ def test_flock_exclusive
       assert_separately(["-rtimeout", "-", regular_file], "#{<<~begin}#{<<~"end;"}")
       begin
         open(ARGV[0], "r") do |f|
+          time_out = !(/mingw|mswin|bccwin/ !~ RUBY_PLATFORM) ? 0.5 : 0.1
           assert_raise(Timeout::Error) do
-            Timeout.timeout(0.1) do
+            Timeout.timeout(time_out) do
               f.flock(File::LOCK_SH)
             end
           end
@@ -1342,7 +1344,8 @@ def test_flock_shared
       assert_separately(["-rtimeout", "-", regular_file], "#{<<~begin}#{<<~"end;"}")
       begin
         open(ARGV[0], "r") do |f|
-          Timeout.timeout(0.1) do
+          time_out = !(/mingw|mswin|bccwin/ !~ RUBY_PLATFORM) ? 0.5 : 0.1
+          Timeout.timeout(time_out) do
             assert(f.flock(File::LOCK_SH))
           end
         end
@@ -1351,7 +1354,8 @@ def test_flock_shared
       begin
         open(ARGV[0], "r+") do |f|
           assert_raise(Timeout::Error) do
-            Timeout.timeout(0.1) do
+            time_out = !(/mingw|mswin|bccwin/ !~ RUBY_PLATFORM) ? 0.5 : 0.1
+            Timeout.timeout(time_out) do
               f.flock(File::LOCK_EX)
             end
           end
